include/quanergy_client_ros/scan_publisher.h

```diff
@@ -0,0 +1,123 @@
+/****************************************************************
+ **                                                            **
+ **  Copyright(C) 2020 Quanergy Systems. All Rights Reserved.  **
+ **  Contact: http://www.quanergy.com                          **
+ **                                                            **
+ ****************************************************************/
+
+#ifndef QUANERGY_CLIENT_ROS_LASERSCAN_PUBLISHER_H
+#define QUANERGY_CLIENT_ROS_LASERSCAN_PUBLISHER_H
+
+#include <mutex>
+
+#include <angles/angles.h>
+#include <quanergy/common/point_hvdir.h>
+#include <pcl/point_cloud.h>
+
+#include <ros/ros.h>
+#include <sensor_msgs/LaserScan.h>
+
+/** \brief SimplePublisher publishes point clouds of type PointT */
+class ScanPublisher
+{
+public:
+  using Cloud = pcl::PointCloud<quanergy::PointHVDIR>;
+  using CloudConstPtr = typename Cloud::ConstPtr;
+
+  ScanPublisher(
+    ros::NodeHandle& nh,
+    const std::string& topic,
+    double frame_rate,
+    float min_range,
+    float max_range,
+    bool use_ros_time = true)
+    : use_ros_time_(use_ros_time)
+  {
+    publisher_ = nh.advertise<sensor_msgs::LaserScan>(nh.resolveName(topic), 10);
+
+    scan_.range_min = min_range;
+    scan_.range_max = max_range;
+    scan_.scan_time = 1.0f / static_cast<float>(frame_rate);
+  }
+
+  void slot(const CloudConstPtr& cloud)
+  {
+    if(!ros::ok() || !cloud || cloud->empty()) return;
+
+    const auto& cloud_deref = *cloud;
+
+    scan_.ranges.reserve(cloud_deref.size());
+    scan_.intensities.reserve(cloud_deref.size());
+    scan_.ranges.clear();
+    scan_.intensities.clear();
+
+    size_t start_i = 0;
+
+    for (size_t i = 0; i < cloud_deref.size(); ++i)
+    {
+      const auto& point = cloud_deref[i];
+
+      if (i > 0 && cloud_deref[i - 1].h > point.h)
```

Comment:
The sensor sometimes has extraneous scan points from the other side of the -pi / pi crossing before it gets to the "real" scan start (at -pi).